aten/src/ATen/native/cuda/Normalization.cuh

```diff
@@ -1468,7 +1581,7 @@ at::Tensor batch_norm_backward_elemt_channels_last_cuda_template(
   if (input.scalar_type() == at::kHalf && weight.defined() && weight.scalar_type() == at::kFloat) {
     AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, input.scalar_type(), "batchnorm_backward_element", [&] {
```

Comment:
pre-existing condition, but why do we need dispatch macro here (that would compile the kernel for all possible dtypes), when `input.scalar_type` is guaranteed to be at::kHalf?
Also, how come there's special accscalar_t handling for half, but not for bfloat16?

Reply:
Fixed BFloat16 handling and cleaned this up a bit.